l3kernel/l3keys.dtx

```diff
@@ -2429,31 +2428,31 @@
 %   afterwards. Note that for speed/simplicity reasons we use a \texttt{tl}
 %   operation to set the \texttt{clist} here!
 %    \begin{macrocode}
-\cs_new_protected:Npn \keys_set_known:nnN #1#2#3
+\cs_new_protected:Npn \keys_set_known:nnN
   {
     \exp_args:No \@@_set_known:nnnnN
-      \l_@@_unused_clist \q_@@_no_value {#1} {#2} #3
+      \l_@@_unused_clist \q_@@_no_value
   }
 \cs_generate_variant:Nn \keys_set_known:nnN { nV , nv , no }
-\cs_new_protected:Npn \keys_set_known:nnnN #1#2#3#4
+\cs_new_protected:Npn \keys_set_known:nnnN #1#2#3
   {
     \exp_args:No \@@_set_known:nnnnN
-      \l_@@_unused_clist {#3} {#1} {#2} #4
+      \l_@@_unused_clist {#3} {#1} {#2}
   }
 \cs_generate_variant:Nn \keys_set_known:nnnN { nV , nv , no }
 \cs_new_protected:Npn \@@_set_known:nnnnN #1#2#3#4#5
   {
     \clist_clear:N \l_@@_unused_clist
     \@@_set_known:nnn {#2} {#3} {#4}
-    \tl_set:Nx #5 { \exp_not:o \l_@@_unused_clist }
+    \cs_set_eq:NN #5 \l_@@_unused_clist
```

Comment:
Please use `\tl_set_eq:NN` now.  Frank clarified that he was complaining about the use of `\cs_...` to set a tl equal to a clist.  We will be documenting explicitly that clists are a subtype of tl, so that `\tl_set_eq:NN \l_my_tl \l_your_clist` is valid code.  The same holds for strings (not used in this pull request I think) and quarks.